q2_american_gut/tests/test_transformer.py

```diff
@@ -0,0 +1,83 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2018, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import unittest
+
+import pandas as pd
+
+import qiime2
+
+from pandas.util.testing import assert_frame_equal
+from q2_american_gut.plugin_setup import QiitaMetadataFormat, QiitaMetadata
+from qiime2.plugin.testing import TestPluginBase
+
+# want metdata to format, format to metadata, qiitameta to meta and meta to qiita
+class TestTransformers(TestPluginBase):
+    package = "q2_american_gut.tests"
+
+    def test_pd_dataframe_to_qiita_metadata_format(self):
+        transformer = self.get_transformer(pd.DataFrame, QiitaMetadataFormat)
+
+        exp_index = pd.Index(['Sample1', 'Sample2'], dtype=object)
+        exp_cols = ['Value']
+        exp = pd.DataFrame([0.55, 0.99], index=exp_index, columns=exp_cols)
+
+        obs = transformer(exp)
+        obs = pd.DataFrame.from_csv(str(obs), sep='\t')
+        assert_frame_equal(exp, obs)
+    
+    def test_qiita_metadata_format_to_pd_dataframe(self):
+        filename = 'qiita-metadata.tsv'
+        _, obs = self.transform_format(QiitaMetadataFormat, pd.DataFrame,
+                                       filename)
+        
+        c1 = ['123.456.789', 'x', 'y', '1.0']
+        c2 = ['123.456.012', 'thing', '0', '2.3']
+        c3 = ['123.xxx.789', float('nan'), 'stuff', 'None']
+        c4 = ['321.xxx.789', float('nan'), 'stuff', 'None']
+        cols = ['#SampleID', 'foo', 'bar', 'baz']
+        exp = pd.DataFrame([c1, c2, c3, c4], columns=cols)
```

Comment:
Shouldn't SampleID be the index column here?

Reply:
it is, it gets set to that in the next line

Reply:
never mind i was looking at the following test

Reply:
the obs dataframe does not come with SampleID set as its index column